erpnext/compliance/doctype/compliance_info/compliance_info.py

```diff
@@ -58,31 +59,50 @@ def get_existing_licensees(license_number, party_type):
 
 
 def validate_license_expiry(doc):
-	if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
-		validate_entity_license("Customer", doc.customer)
-	elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
-		validate_entity_license("Supplier", doc.supplier)
-	elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
-		validate_entity_license("Customer", doc.party_name)
-
-
-@frappe.whitelist()
-def validate_entity_license(party_type, party_name):
-	license_record = get_default_license(party_type, party_name)
-	if not license_record:
+	validate_docinfo = validate_doc_license(doc)
+	if validate_docinfo:
+		if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
+			validate_entity_license(doc, "Customer", doc.customer)
+		elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
+			validate_entity_license(doc, "Supplier", doc.supplier)
+		elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
+			validate_entity_license(doc, "Customer", doc.party_name)
+
+def validate_doc_license(doc):
+	"""Check if any compliance item available in the items table"""
+	is_compliance_item = False
+	compliance_items = frappe.get_all('Compliance Item', fields=['item_code'])
+	if not compliance_items:
 		return
 
-	license_expiry_date, license_number = frappe.db.get_value(
-		"Compliance Info", license_record, ["license_expiry_date", "license_number"])
+	for item in (doc.get("items") or []):
+		compliance_item = next((data for data in compliance_items if data.get("item_code") == item.get("item_code")), None)
+		if compliance_item:
+			is_compliance_item = True
+	return is_compliance_item
 
-	if not license_expiry_date:
-		frappe.msgprint(_("We could not verify the status of license number {0}, Proceed with Caution.").format(
-			frappe.bold(license_number)))
-	elif license_expiry_date < getdate(nowdate()):
-		frappe.msgprint(_("Our records indicate {0}'s license number {1} has expired on {2}, Proceed with Caution.").format(
-			frappe.bold(party_name), frappe.bold(license_number), frappe.bold(license_expiry_date)))
-
-	return license_record
+@frappe.whitelist()
+def validate_entity_license(doc, party_type, party_name):
```

Comment:
@vishdha, your function is now doing 2 things:

- taking a document, and validating that document's items to be compliance
- taking a party document, and validating its license expiry

You need to do these with smaller functions so that they're easy to use, and handle them in your whitelisted function instead.

Something like:

```python
@frappe.whitelist()
get_entity_license(doc, party_type, party_name):
	# get the default license for the given party
	license = validate_entity_license(party_type, party_name)

	# show a warning if a license exists and is expired, and only if compliance items are present
	is_compliance = validate_doc_compliance(doc)
	if is_compliance:
		# show warnings
```